lib/src/data_table_2.dart

```diff
@@ -362,23 +377,34 @@ class DataTable2 extends DataTable {
         child: DropdownButtonHideUnderline(child: label),
       ),
     );
-    if (onTap != null) {
+    if (onTap != null ||
+        onDoubleTap != null ||
+        onLongPress != null ||
+        onTapDown != null ||
+        onTapCancel != null) {
       label = InkWell(
         onTap: onTap,
-        child: label,
+        onDoubleTap: onDoubleTap,
+        onLongPress: onLongPress,
+        onTapDown: onTapDown,
+        onTapCancel: onTapCancel,
         overlayColor: overlayColor,
+        child: label,
       );
     } else {
       label = GestureDetector(
         child: TableRowInkWell(
-            child: label,
-            overlayColor: overlayColor,
-            onTap: onRowTap == null
-                ? onSelectChanged
-                : () {
-                    onRowTap();
-                    onSelectChanged();
-                  }),
+          child: label,
+          overlayColor: overlayColor,
+          onTap: onRowTap == null
+              ? onSelectChanged
+              : () {
+                  onRowTap();
+                  onSelectChanged();
```

Comment:
I kept this unchanged, but I did not call `onSelectChanged()` in `onDoubleTap` and `onLongPress` below.

Reply:
Not sure what you mean under "this unchanged" though it makes sense to call onSelectChanged only for single tap